File: AntonDanik.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int games;
    string result;
    cin >> games >> result;
    int anton = 0, danik = 0;
    for (int i=0; i<games; i++) {
        if (result[i] == 'A')
            anton++;
        else if (result[i] == 'D')
            danik++;
    }
    if (anton > danik)
        cout << "Anton\n";
    else if (danik > anton)
        cout << "Danik\n";
    else if (anton == danik)
        cout << "Friendship\n";

    return 0;
}

File: Python/AntonDanik.py
def main():
    games = int(input())
    result = input()
    anton, danik = 0, 0
    for i in range(games):
        if result[i] == 'A':
            anton += 1
        elif result == 'D':
            danik += 1
    if anton > danik:
        print("Anton\n")
    elif anton < danik:
        print("Danik\n")
    elif anton == danik:
        print("Friendship\n")


if __name__ == "__main__":
    main()
